src/Adapter/Order/CommandHandler/AddOrderFromBackOfficeHandler.php

```diff
@@ -89,13 +108,19 @@ public function handle(AddOrderFromBackOfficeCommand $command)
                 $cart->secure_key
             );
         } catch (Exception $e) {
+            $this->contextStateManager->restoreContext();
+
             throw new OrderException('Failed to add order. ' . $e->getMessage(), 0, $e);
         }
```

Comment:
You can deal with all the use cases with:
```php
} finally {
    $this->contextStateManager->restoreContext();
}
```

This will leave you with only one call

Reply:
Done